apps/etl-func/src/domain/message-status.ts

```diff
@@ -23,27 +23,48 @@ const statusEnum = z.enum([
   "REJECTED",
 ]);
 
-const rejectedMessageStatusSchema = z.object({
-  fiscalCode: fiscalCodeSchema,
-  isArchived: z.boolean().default(false),
-  isRead: z.boolean().default(false),
-  messageId: z.string().ulid(),
-  rejection_reason: rejectionReasonSchema,
-  status: statusEnum.extract(["REJECTED"]),
-  updatedAt: z.number(),
-});
-
-const notRejectedMessageStatusSchema = z.object({
-  fiscalCode: fiscalCodeSchema,
-  isArchived: z.boolean().default(false),
-  isRead: z.boolean().default(false),
-  messageId: z.string().ulid(),
-  status: statusEnum.exclude(["REJECTED"]),
-  updatedAt: z.number(),
-});
+export const messageStatusSchema = z
+  .object({
+    fiscalCode: fiscalCodeSchema,
+    id: z.string().min(1),
+    isArchived: z.boolean().default(false),
+    isRead: z.boolean().default(false),
+    messageId: z.string().ulid(),
+    updatedAt: z.number(),
+    version: z.number().gte(0),
+  })
+  .and(
+    z.discriminatedUnion("status", [
+      z.object({
+        status: statusEnum.exclude(["REJECTED"]),
+      }),
+      z.object({
+        rejection_reason: rejectionReasonSchema,
+        status: statusEnum.extract(["REJECTED"]),
+      }),
+    ]),
+  );
+export type MessageStatus = z.TypeOf<typeof messageStatusSchema>;
 
-export const messageStatusSchema = z.discriminatedUnion("status", [
-  notRejectedMessageStatusSchema,
-  rejectedMessageStatusSchema,
+export const messageStatusEventOperationSchema = z.enum([
+  "CREATE",
+  "UPDATE",
+  "DELETE",
 ]);
-export type MessageStatus = z.TypeOf<typeof messageStatusSchema>;
+export type MessageStatusEventOperation = z.TypeOf<
+  typeof messageStatusEventOperationSchema
+>;
```

Comment:
is `messageStatusEventOperationSchema` (what a long name!) used? if not, it can be removed

Reply:
[5aef172](https://github.com/pagopa/io-messages/pull/62/commits/5aef1727138b1f1535c2b0878def6d3eb8294319)